securebanking-openbanking-uk-rs-simulator/src/main/java/com/forgerock/securebanking/openbanking/uk/rs/api/obie/payment/v3_0/domesticpayments/DomesticPaymentsApiController.java

```diff
@@ -0,0 +1,130 @@
+        log.trace("Converted to: '{}'", frDomesticPayment);
+        FRDomesticPaymentSubmission frPaymentSubmission = FRDomesticPaymentSubmission.builder()
+                // TODO - openbanking-aspsp uses the consent id - is this for convenience? Could it be a problem?
+                .id(frDomesticPayment.getData().getConsentId())
+                .domesticPayment(frDomesticPayment)
+                .status(PENDING)
+                .created(new DateTime())
+                .updated(new DateTime())
+                .idempotencyKey(xIdempotencyKey)
+                .obVersion(VersionPathExtractor.getVersionFromPath(request))
+                .build();
+        frPaymentSubmission = new IdempotentRepositoryAdapter<>(domesticPaymentSubmissionRepository)
+                .idempotentSave(frPaymentSubmission);
+        return ResponseEntity.status(HttpStatus.CREATED).body(responseEntity(frPaymentSubmission));
+    }
+
+    @Override
+    public ResponseEntity getDomesticPaymentsDomesticPaymentId(String domesticPaymentId,
+                                                               String xFapiFinancialId,
+                                                               String authorization,
+                                                               DateTime xFapiCustomerLastLoggedTime,
+                                                               String xFapiCustomerIpAddress,
+                                                               String xFapiInteractionId,
+                                                               String xCustomerUserAgent,
+                                                               HttpServletRequest request,
+                                                               Principal principal
+    ) {
+        Optional<FRDomesticPaymentSubmission> isPaymentSubmission = domesticPaymentSubmissionRepository.findById(domesticPaymentId);
+        if (!isPaymentSubmission.isPresent()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Payment submission '" + domesticPaymentId + "' can't be found");
+        }
+        FRDomesticPaymentSubmission frPaymentSubmission = isPaymentSubmission.get();
+        return ResponseEntity.ok(responseEntity(frPaymentSubmission));
+    }
+
+    private OBWriteDomesticResponse1 responseEntity(FRDomesticPaymentSubmission frPaymentSubmission) {
+        return new OBWriteDomesticResponse1()
+                .data(new OBWriteDataDomesticResponse1()
+                        .domesticPaymentId(frPaymentSubmission.getId())
+                        .initiation(toOBDomestic1(frPaymentSubmission.getDomesticPayment().getData().getInitiation()))
+                        .creationDateTime(frPaymentSubmission.getCreated())
+                        .statusUpdateDateTime(frPaymentSubmission.getUpdated())
+                        .status(toOBTransactionIndividualStatus1Code(frPaymentSubmission.getStatus()))
```

Comment:
There are a few subtle changes here compared with openbanking-aspsp. The status is saved against the payment submission (as an instance of `PaymentSubmissionStatus`) and we map it to the required API version object. Note that openbanking-aspsp maps the consent status here, which I don't think is right.